### PR TITLE
Bug Fix for the Hips not translating correctly when Hips and Root have the same location

### PR DESCRIPTION
**Pull request by amantley**
Manuscript cases are here:
https://highfidelity.manuscript.com/f/cases/20524/avatar-with-armature-root-up-near-hips-mutes-the-hips-translation

https://highfidelity.manuscript.com/f/cases/20619/

Fixed the hips translation in the animations, this was related to scaling the translation that happens in AnimClip.cpp

Fixed the ik problem caused by the hips and the root being at the same height. This was caused by a root node being included in the accumulators in AnimInverseKinematics.cpp.  When the joint chains are created they include all joints until parent == -1.    

### TEST PLAN
1)Try these avatars from the case 
a) https://hifi-content.s3.us-west-1.amazonaws.com/davidback/avatars/PinnedAtHips/MaleInBlueJean/avatar.fst
b) https://hifi-content.s3.us-west-1.amazonaws.com/davidback/avatars/PinnedAtHips/UmberHulk/avatar.fst
c) http://hifi-content.s3.amazonaws.com/angus/rootHipsBug/valkyrie/avatar.fst
d) http://hifi-content.s3.amazonaws.com/angus/rootHipsBug/lexichi/avatar.fst

Press space bar and see that they jump and land with the hips swaying.
They should all animate fairly normally now.  Blue Jean guy has strange parenting of the legs so he looks a bit stiffer.  

2) go into hmd mode and see that the ik is working
They should all bend their knees and the arms should follow the hand controllers.
**Note**: in first person, Hulk and Blue Jeans's arms disappear in first person.  This is a separate bug that is caused by the arms being children of the neck instead of spine2.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17695/HighFidelity-Beta-PR15046-580827f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17695/HighFidelity-Beta-Interface-PR15046-580827f.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17695/HighFidelity-Beta-PR15046-580827f.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17695/HighFidelity-Quest-Beta-PR15046-580827f.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17695/HighFidelity-Beta-PR15046-580827f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17695/HighFidelity-Beta-Interface-PR15046-580827f.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17713/HighFidelity-Beta-PR15046-dd35a47.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17713/HighFidelity-Quest-Beta-PR15046-dd35a47.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17713/HighFidelity-Beta-PR15046-dd35a47.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17713/HighFidelity-Beta-Interface-PR15046-dd35a47.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17713/HighFidelity-Beta-PR15046-dd35a47.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17713/HighFidelity-Beta-Interface-PR15046-dd35a47.exe

File: libraries/animation/src/AnimClip.cpp [hunk 1]
```diff
@@ -151,9 +151,30 @@ void AnimClip::copyFromNetworkAnim() {
...
-                    boneLengthScale = glm::length(avatarDefaultPose.trans()) / glm::length(animZeroTrans);
+
+                const int avatarHipsParentIndex = _skeleton->getParentIndex(_skeleton->nameToJointIndex("Hips"));
+                if (avatarJointIndex == _skeleton->nameToJointIndex("Hips") && (avatarHipsParentIndex != -1)) {
+
+                    const AnimPose& animHipsAbsoluteDefaultPose = animSkeleton.getAbsoluteDefaultPose(animSkeleton.nameToJointIndex("Hips"));
+                    const AnimPose& avatarHipsAbsoluteDefaultPose = _skeleton->getAbsoluteDefaultPose(_skeleton->nameToJointIndex("Hips"));
+                    const float animSkeletonHeight = fabsf(glm::length(animHipsAbsoluteDefaultPose.trans()));
+
+                    // this corrects for the common case where the model root is scaled
```
Comment (hyperlogic) [inline libraries/animation/src/AnimClip.cpp]: I don't think we need to do this weird inverse thing.  Instead just multiply the absolute hips translation by the model.offset.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17774/HighFidelity-Beta-PR15046-e3af933.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17774/HighFidelity-Quest-Beta-PR15046-e3af933.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17774/HighFidelity-Beta-PR15046-e3af933.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17774/HighFidelity-Beta-Interface-PR15046-e3af933.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17774/HighFidelity-Beta-PR15046-e3af933.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17774/HighFidelity-Beta-Interface-PR15046-e3af933.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17798/HighFidelity-Beta-PR15046-217145f.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17798/HighFidelity-Quest-Beta-PR15046-217145f.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17798/HighFidelity-Beta-PR15046-217145f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17798/HighFidelity-Beta-Interface-PR15046-217145f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17798/HighFidelity-Beta-PR15046-217145f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17798/HighFidelity-Beta-Interface-PR15046-217145f.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17812/HighFidelity-Beta-PR15046-a11c4bf.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17812/HighFidelity-Quest-Beta-PR15046-a11c4bf.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17812/HighFidelity-Beta-PR15046-a11c4bf.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17812/HighFidelity-Beta-Interface-PR15046-a11c4bf.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17817/HighFidelity-Beta-PR15046-c046b8f.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17817/HighFidelity-Quest-Beta-PR15046-c046b8f.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17817/HighFidelity-Beta-PR15046-c046b8f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17817/HighFidelity-Beta-Interface-PR15046-c046b8f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17817/HighFidelity-Beta-PR15046-c046b8f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17817/HighFidelity-Beta-Interface-PR15046-c046b8f.exe

**Pull request merged**